src/python/WMCore/WMRuntime/Watchdog.py

```diff
@@ -78,6 +81,37 @@ def setupMonitors(self, task, wmbsJob):
                 #This should be a config section
                 monitorArgs = getattr(task.data.watchdog, monitor)
                 args = monitorArgs.dictionary_()
+                # Scale resources according to the HTCondor runtime environment.
+                origCores = 1
+                try:
+                    origCores = int(task.data.application.multicore.numberOfCores)
+                except AttributeError:
+                    print("No value for numberOfCores.  Assuming 1")
+                resources = {'cores': origCores}
+                origMaxRSS = args.get('maxRSS')
+                if origMaxRSS:
+                    resources['memory'] = origMaxRSS
+                # Actually parses the HTCondor runtime
+                resizeResources(resources)
+                # HTCondor doesn't explicitly scale VSize; hence we scale it
+                # according to the scaling of maxRSS between the workflow and runtime.
+                if origMaxRSS and ('maxVSize' in args):
+                    args['maxVSize'] *= resources['memory'] / origMaxRSS
```

Comment:
I just had an example where the maxVSize was set to 30GB for single core, the job matched to 2 cpus and this value was reduced to ~3.7GB, a very likely value to be reached by the job.

I suggest we change it taking into consideration the `coreScaling` value, something like:
```
if 'maxVSize' in args:
  args['maxVSize'] *= coreScaling
```